src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -270,13 +271,15 @@
     public SQLServerBulkCopy(Connection connection) throws SQLServerException {
         loggerExternal.entering(loggerClassName, "SQLServerBulkCopy", connection);
 
-        if (null == connection || !(connection instanceof SQLServerConnection)) {
+        if (null == connection || !(connection instanceof ISQLServerConnection)) {
             SQLServerException.makeFromDriverError(null, null,
                     SQLServerException.getErrString("R_invalidDestConnection"), null, false);
         }
 
         if (connection instanceof SQLServerConnection) {
             this.connection = (SQLServerConnection) connection;
+        } else if (connection instanceof SQLServerConnectionPoolProxy) {
+            this.connection = ((SQLServerConnectionPoolProxy) connection).getWrappedConnection();
```

Comment:
should we also be checking for XAConnection here?

Reply:
For both XA and Pooled Connections, `getConnection()` returns connection object which is instance of `SQLServerConnectionPoolProxy`, hence this check applies to both parent connection types.